message_passing/update_with_start/lazy_initialization/workflows.py

```diff
@@ -0,0 +1,51 @@
+from dataclasses import dataclass
+from decimal import Decimal
+from typing import List, Tuple
+
+from temporalio import workflow
+from temporalio.exceptions import ApplicationError
+
+from message_passing.update_with_start.lazy_initialization.activities import (
+    ShoppingCartItem,
+    get_price,
+)
+
+
+@dataclass
+class FinalizedOrder:
+    id: str
+    items: List[Tuple[ShoppingCartItem, str]]
+    total: str
+
+
+@workflow.defn
+class ShoppingCartWorkflow:
+    def __init__(self):
+        self.items: List[Tuple[ShoppingCartItem, Decimal]] = []
+        self.order_submitted = False
+
+    @workflow.run
+    async def run(self) -> FinalizedOrder:
+        await workflow.wait_condition(
+            lambda: workflow.all_handlers_finished() and self.order_submitted
+        )
+        return FinalizedOrder(
+            id=workflow.info().workflow_id,
+            items=[(item, str(price)) for item, price in self.items],
+            total=str(sum(price for _, price in self.items) or Decimal("0.00")),
+        )
+
+    @workflow.update
+    async def add_item(self, item: ShoppingCartItem) -> str:
+        price = await get_price(item)
```

Comment:
I think you want this to be an activity invocation

Reply:
Thanks! So (a) too much Typescript, and (b) this is a good example of a mistake that an AI linter could catch but traditional analysis wouldn't.

Reply:
Fixed